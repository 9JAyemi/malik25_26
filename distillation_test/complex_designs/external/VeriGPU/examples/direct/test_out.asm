# test writing to 1000 as stdout

li x31, 1000000

li x1, 123
sw x1, 0(x31)

li x1, 44
sw x1, 0(x31)

li x1, 55
sw x1, 0(x31)

# test out

out 123
out 789
out 0x32
out 0x44
out 50
out 60

; test outloc

outloc 600; 240
outloc 604
outloc 600
outloc 604
outloc 656
outloc 660
outloc 664
outloc 668

halt

# test halt, by putting instructions after it

out 70
out 80


location 600:
word 0x0000dead
word 0x0000beef

location 656:
word 0x00001111
word 0x00002222
word 15
word 21
