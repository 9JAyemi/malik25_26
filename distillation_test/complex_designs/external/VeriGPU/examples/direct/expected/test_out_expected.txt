out 0 00000000000000000000000001111011 0000007b 123
out 1 00000000000000000000000000101100 0000002c 44
out 2 00000000000000000000000000110111 00000037 55
out 3 00000000000000000000000001111011 0000007b 123
out 4 00000000000000000000001100010101 00000315 789
out 5 00000000000000000000000000110010 00000032 50
out 6 00000000000000000000000001000100 00000044 68
out 7 00000000000000000000000000110010 00000032 50
out 8 00000000000000000000000000111100 0000003c 60
out 9 00000000000000001101111010101101 0000dead 57005
out 10 00000000000000001011111011101111 0000beef 48879
out 11 00000000000000001101111010101101 0000dead 57005
out 12 00000000000000001011111011101111 0000beef 48879
out 13 00000000000000000001000100010001 00001111 4369
out 14 00000000000000000010001000100010 00002222 8738
out 15 00000000000000000000000000001111 0000000f 15
out 16 00000000000000000000000000010101 00000015 21
